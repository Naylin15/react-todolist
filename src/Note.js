import React, { Component } from 'react';
import { Checkbox, Grid, Header, Dropdown } from 'semantic-ui-react';
// import DropdownOptions from './displayNote/DropdownOptions';
import DropdownCategory from './displayNote/DropdownCategory';

function OnlyNote(props) {
    console.log(props);
    const singleNote = props.notesArray.map((note, index) => {
       return (
            <Grid.Row centered columns={3} key={index}>
                <Grid.Column width={2}>
                    <Checkbox />
                </Grid.Column>
                <Grid.Column>
                    <Header as='h2'>{note.title}</Header>
                </Grid.Column>
                <Grid.Column>
                    <DropdownCategory />
                    <Dropdown icon='ellipsis vertical'>
                        <Dropdown.Menu>
                            <Dropdown.Item text='Delete' onClick={() => this.removeNote(index, props.notesArray)}/>
                            <Dropdown.Item text='Edit' />
                        </Dropdown.Menu>
                    </Dropdown>
                </Grid.Column>
            </Grid.Row>
       )
     })
    
    return singleNote

}


class Note extends Component {
    constructor(props) {
        super(props);
        this.removeNote = this.removeNote.bind(this);
        // the notesArray which is being passed as props is now part of the state
        this.state = {
            notesArray: this.props.notesArray,
        }
    }

    removeNote(id, notes) {
        let newArray = notes;
        newArray.splice(id, 1);
        console.log(newArray);
        //     // return <OnlyNote notesArray={newArray}/>
    }

    render() {
        return (
            <Grid>
                <OnlyNote notesArray={this.state.notesArray} removeNote={this.removeNote}/>
            </Grid>
        )
    }
}

export default Note