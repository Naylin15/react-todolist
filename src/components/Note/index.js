import React from 'react';
import { Grid } from 'semantic-ui-react';
import SingleNote from '../SingleNote';

class Note extends React.Component {
    constructor(props) {
        super(props);
        this.removeNote = this.removeNote.bind(this);
        this.editNote = this.editNote.bind(this);
        this.saveChanges = this.saveChanges.bind(this);
        this.checkNote = this.checkNote.bind(this);
        this.addCategory = this.addCategory.bind(this);
        // the notesArray which is being passed as props is now part of the state
        this.state = {
            notesArray: this.props.notesArray,
        }
    }

    checkNote(id, e, data) {
        const finished = document.getElementById(id);
        if (data.checked) {
           finished.className = "centered three column row finished";
        } else {
           finished.className = "centered three column row"
        }
    }

    addCategory(id, e, data) {
        let newArray = [...this.state.notesArray];
        newArray[id].category = data.value;
        this.setState({notesArray: newArray});
        this.props.updateState(newArray);
    }

    removeNote(id) {
        let newArray = [...this.state.notesArray];
        newArray.splice(id, 1);
        this.setState({notesArray: newArray});
        this.props.updateState(newArray);
    }

    editNote(id) {
        let newArray = [...this.state.notesArray];
        newArray[id].edit = true;
        this.setState({notesArray: newArray});
        this.props.updateState(newArray);
    }
    
    saveChanges(id, newValue) {
        let newArray = [...this.state.notesArray];
        newArray[id].title = newValue;
        newArray[id].edit = false;
        this.setState({notesArray: newArray});
        this.props.updateState(newArray);
    }

    render() {
        return (
            <Grid>
                {this.state.notesArray.map((note, index) =>  
                    <SingleNote
                        key={index}
                        noteId={index}
                        note={note}
                        editNote={this.editNote}
                        removeNote={this.removeNote}
                        saveChanges={this.saveChanges}
                        checkNote={this.checkNote}
                        addCategory={this.addCategory}
                    />
                )}
            </Grid>
        )
    }
}

export default Note