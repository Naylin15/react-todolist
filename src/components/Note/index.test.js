import React from 'react';
import renderer from 'react-test-renderer'
import Note from './index';

it('renders with notes array', () => {
    const tree = renderer
      .create(<Note notesArray={[{title: 'abc'}]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

