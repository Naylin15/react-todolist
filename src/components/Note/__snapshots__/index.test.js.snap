// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with notes array 1`] = `
<div
  className="ui grid"
>
  <div
    className="centered three column row"
    id={0}
  >
    <div
      className="two wide column"
    >
      <div
        className="ui fitted checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
        />
        <label />
      </div>
    </div>
    <div
      className="column"
    >
      <h2
        className="ui header"
        id="note-0"
      >
        abc
      </h2>
    </div>
    <div
      className="column"
    >
      <div
        aria-expanded={false}
        className="ui dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <i
          aria-hidden="true"
          className="filter icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            <span
              className="text"
            >
              Important
            </span>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            <span
              className="text"
            >
              Tomorrow
            </span>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            <span
              className="text"
            >
              Later
            </span>
          </div>
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <i
          aria-hidden="true"
          className="ellipsis vertical icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            <span
              className="text"
            >
              Delete
            </span>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="option"
          >
            <span
              className="text"
            >
              Edit
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
